src/hash.h

```diff
@@ -122,8 +122,8 @@ class CHashWriter
     const int nVersion;
 public:
 
-    CHashWriter() : CHashWriter(SER_GETHASH, 0) {}
-    CHashWriter(int nVersionIn) : CHashWriter(SER_GETHASH, nVersionIn) {}
+    CHashWriter() : CHashWriter(0, 0) {}
```

Comment:
Can you use SER_NETWORK instead here? 0 is not a valid serialization type, even if it's not actually used anywhere.

Reply:
Reviewing calls to `GetType` after these changes, it seems like there are only 2 calls that affect the output, both testing `GetType` for `SER_DISK` in `CAddress::SerializationOp`. Is it possible to remove those? If so we could get right of `GetType` entirely, which seems like a big win.

If not, it seems like SER_NETWORK would not cause negative effects, but it would be a bit misleading for those cases where the serialization is not propagated to the network.

Reply:
Relatedly: #13558

Reply:
As of #13560, I'm pretty confident that `GetType` can be removed across the board.

Reply:
Switched to `SER_NETWORK` - it's equivalent to `0` now as only `SER_DISK` has conditional behavior associated with it.